src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,137 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public static class Node<T> {
+        private T item;
+        private Node<T> prev;
+        private Node<T> next;
+
+        public Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.item = value;
+            this.next = next;
+        }
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (head == null && tail == null) {
+            Node<T> newNod = new Node<>(null, value, null);
+            head = newNod;
+            tail = newNod;
+        } else {
+            Node<T> someNod = new Node<>(tail, value, null);
+            tail.next = someNod;
+            tail = someNod;
+        }
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (index == 0) {
+            Node<T> newNode = new Node<>(null, value, head);
+            head.prev = newNode;
+            head = newNode;
+        } else {
+            Node<T> perAddNod = search(index);
+            Node<T> addNode = new Node<>(perAddNod.prev, value, perAddNod);
+            perAddNod.prev.next = addNode;
+            perAddNod.prev = addNode;
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T t : list) {
+            add(t);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        Node<T> search = search(index);
+        return search.item;
```

Comment:
I'm not sure, but maybe you can return search(index).item

Reply:
IDEA don't like this idea)

Reply:
:D